ui/tests/integration/components/tools/wrap-test.js

```diff
@@ -104,6 +105,53 @@ module('Integration | Component | tools/wrap', function (hooks) {
     await click(TS.submit);
   });
 
+  test('it toggles between views and preserves input data', async function (assert) {
+    assert.expect(4);
+    await this.renderComponent();
+    await codemirror().setValue(this.wrapData);
+    assert.dom('[data-test-component="json-editor-title"]').hasText('Data to wrap (json-formatted)');
+    await click('[data-test-toggle-input="json"]');
+    assert.dom('[data-test-component="json-editor-title"]').doesNotExist();
+    await click('[data-test-toggle-input="json"]');
+    assert.dom('[data-test-component="json-editor-title"]').exists();
+    assert.strictEqual(
+      codemirror().getValue(' '),
+      `{   \"foo": \"bar" }`, // eslint-disable-line no-useless-escape
+      'json editor has original data'
+    );
+  });
+
+  test('it submits from kv view', async function (assert) {
+    assert.expect(6);
+
+    const flashSpy = sinon.spy(this.owner.lookup('service:flash-messages'), 'success');
+
+    this.server.post('sys/wrapping/wrap', (schema, { requestBody, requestHeaders }) => {
+      const payload = JSON.parse(requestBody);
+      assert.propEqual(payload, JSON.parse(this.wrapData), `payload contains data: ${requestBody}`);
+      assert.strictEqual(requestHeaders['X-Vault-Wrap-TTL'], '30m', 'request header has default wrap ttl');
+      return {
+        wrap_info: {
+          token: this.token,
+          accessor: '5yjKx6Om9NmBx1mjiN1aIrnm',
+          ttl: 1800,
+          creation_time: '2024-06-07T12:02:22.096254-07:00',
+          creation_path: 'sys/wrapping/wrap',
+        },
+      };
+    });
+
+    await this.renderComponent();
+    await codemirror().setValue(this.wrapData);
+    await click('[data-test-toggle-input="json"]');
```

Comment:
After toggling, before submitting can we input one more row of data? I want to make sure the tracked property is updated properly. It may be useful to input a multi line string here as well! That way we're also making sure to test for the original request.